learn/experimental/vector_search.mdx

```diff
@@ -30,39 +30,151 @@ curl \
   }'
 ```
 
-### Generate vector embeddings using a third-party tool
+### Generate vector embeddings
 
-To use vector search, you must generate embeddings with a third-party tool such as [OpenAI](https://openai.com/), [HuggingFace](https://huggingface.co/), and [Cohere.AI](https://cohere.com/).
+To use vector search, first configure the `embedders` index setting. You may configure multiple embedders for an index. 
 
-<Capsule intent="note" title="What is an embedding?">
-Vector search relies on embeddings. Vector embeddings are a way of representing data that identify similarities in meaning and context.
+Embedders generate vector data from your documents. Meilisearch natively supports [OpenAI](https://openai.com/) and [HuggingFace](https://huggingface.co/) embedders. 
+
+It is also possible to use custom embedders. In this case, you must generate the embeddings manually and include them as a field in your documents.
+
+#### Generate auto-embeddings with OpenAI 
+
+Use the `embedders` index setting of the [update `/settings` endpoint](/reference/api/settings) to configure one or more embedders for an index:
+
+```sh
+curl \
+  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "embedders": {
+      "default": {
+        "source":  "openAI",
+        "apiKey": "anOpenAiApiKey",
+        "model": "text-embedding-ada-002",
+        "documentTemplate": "A movie titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"
+      }
+    }
+  }'
+```
+
+It is mandatory to pass an OpenAI API key through the `OPENAI_API_KEY` environment variable or the `apiKey` field when using an OpenAI embedder. [Generate an API key](https://platform.openai.com/api-keys) from your OpenAI account. Use [tier 2 keys](https://platform.openai.com/docs/guides/rate-limits/usage-tiers?context=tier-two) or above for optimal performance.
+
+`documentTemplate` is an optional field you can use to customize the data you send to the embedder. It is highly recommended you configure a custom template for your documents.
+
+<Capsule intent="tip" title="`documentTemplate` usage">
+`documentTemplate` must be a [Liquid template](https://shopify.github.io/liquid/). Use `{{ doc.attribute }}` to access the `attribute` field value of your documents. Any field you refer to in this way must exist in all documents or an error will be raised at indexing time.
+
+For best results, use short strings indicating the type of document in that index, only include highly relevant document fields, and truncate long fields.  
 </Capsule>
 
-### Add embeddings to Meilisearch
+#### Generate auto-embeddings with HuggingFace 
+
+The HuggingFace computes embeddings locally. This is a resource-intensive operation and might affect indexing performance.
```

Comment:
```suggestion
HuggingFace computes embeddings locally. This is a resource-intensive operation and might affect indexing performance.
```

Reply:
Alternatively "The HuggingFace embedder computes"?

Reply:
Yeap, thanks for spotting it. I did mean to write "The HuggingFace embedder" 😅